File: android/src/main/java/com/zh4dev/niimbot_print/Model/PrinterConfigurationModel.java
package com.zh4dev.niimbot_print.Model;

import com.google.gson.Gson;

public class PrinterConfigurationModel {

    public int printModel;
    public int printDensity;
    public float printMultiple;
    public PrinterConfigurationModel() {
    }
    public PrinterConfigurationModel(int printModel, int printDensity, float printMultiple) {
        this.printModel = printModel;
        this.printDensity = printDensity;
        this.printMultiple = printMultiple;}

    public String toMap() {
        return new Gson().toJson(this);
    }

    public PrinterConfigurationModel fromMap(String json) {
        return new Gson().fromJson(json, PrinterConfigurationModel.class);
    }

}

File: ios/Classes/PrinterConfigurationModel.m
#import "PrinterConfigurationModel.h"

@implementation PrinterConfigurationModel

- (instancetype)init {
    self = [super init];
    if (self) {
        _printModel = 0;
        _printDensity = 0;
        _printMultiple = 0.0;
    }
    return self;
}

- (instancetype)initWithPrintModel:(int)printModel printDensity:(int)printDensity printMultiple:(float)printMultiple {
    self = [super init];
    if (self) {
        _printModel = printModel;
        _printDensity = printDensity;
        _printMultiple = printMultiple;
    }
    return self;
}

- (NSString *)toJson {
    NSError *error = nil;
    NSDictionary *dictionary = @{
        @"printModel": @(self.printModel),
        @"printDensity": @(self.printDensity),
        @"printMultiple": @(self.printMultiple)
    };
    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dictionary options:NSJSONWritingPrettyPrinted error:&error];

    if (error) {
        return @"";
    } else {
        return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
    }
}

+ (PrinterConfigurationModel *)fromJson:(NSString *)json {
    NSError *error = nil;
    NSData *jsonData = [json dataUsingEncoding:NSUTF8StringEncoding];
    NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:jsonData options:kNilOptions error:&error];

    if (error) {
        return [[PrinterConfigurationModel alloc] init];
    } else {
        return [[PrinterConfigurationModel alloc] initWithPrintModel:[dictionary[@"printModel"] intValue]
                                                       printDensity:[dictionary[@"printDensity"] intValue]
                                                     printMultiple:[dictionary[@"printMultiple"] floatValue]];
    }
}

@end
